### PR TITLE
FindSortRemind

### PR DESCRIPTION
**Pull request by KevinAlvarez7**

### CODE REVIEW

File: src/main/java/duke/models/locker/Locker.java [hunk 1]
```diff
@@ -196,4 +203,79 @@ public boolean equals(Object other) {
...
+
+    public boolean findExpiryDate(LocalDate now) {
+
+        try {
+            String endDate = this.usage.getEndDate().getDate();
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-uuuu");
+            LocalDate expiryDate = LocalDate.parse(endDate,formatter);
+            int daysBetween = (int) DAYS.between(now, (Temporal) expiryDate);
+
+            if (daysBetween <= 30) {
```
Comment (Tejasvas123) [inline src/main/java/duke/models/locker/Locker.java]: we decide 7 days right?

File: src/main/java/duke/logic/commands/RemindersCommand.java [hunk 2]
```diff
@@ -2,12 +2,59 @@
...
     @Override
     public void execute(LockerList lockerList, Ui ui, Storage storage) throws DukeException {
 
+
+        LocalDate now = LocalDate.now();
+        String dateNow = now.format(DateTimeFormatter.ofPattern("dd-MM-uuuu"));
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-uuuu");
+        LocalDate localDateNow = LocalDate.parse(dateNow,formatter);
+
+        FindCommand.FindLocker unauthorisedLocker = new FindCommand.FindLocker();
```
Comment (Tejasvas123) [inline src/main/java/duke/logic/commands/RemindersCommand.java]: can explain me the use of using findLocker here?

File: docs/USER_GUIDE.adoc [hunk 3]
```diff
@@ -120,6 +120,50 @@ Examples:
 
 * `addLocker s/123 a/Com1 Level2 z/A`
 
+<<<<<<< HEAD
+==== Finding and searching for lockers: `find`
```
Comment (Tejasvas123) [inline docs/USER_GUIDE.adoc]: merge conflicts not fixed for this haha

File: src/main/java/duke/logic/commands/RemindersCommand.java [hunk 4]
```diff
@@ -2,12 +2,59 @@
...
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+
 public class RemindersCommand extends Command {
+
+    public static final String COMMAND_WORD = "reminders";
+
+    public RemindersCommand() {
```
Comment (Tejasvas123) [inline src/main/java/duke/logic/commands/RemindersCommand.java]: can remove empty constructors

File: src/main/java/duke/logic/commands/RemindersCommand.java (hunk 2 above)
Comment (Tejasvas123) [reply src/main/java/duke/logic/commands/RemindersCommand.java]: to search for unauthorised lockers/broken lockers you can just to the following
lockerList.getLockerList().stream(p -> p.getTag().equals(new Tag("unauthorized")
similarly  for broken.
Dont actually need findLocker

**Comment by KevinAlvarez7** [comment]
I've made the changes necessary.

**Review by Tejasvas123 — APPROVED** [review]
Approved! Good job!

**Pull request merged**